x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/aggregate/package-info.java

```diff
@@ -0,0 +1,168 @@
+ *         and add a test for the function you want to write. These files are roughly themed but there
+ *         isn't a strong guiding principle in the organization.
+ *     </li>
+ *     <li>
+ *         Rerun the {@code CsvTests} and watch your new test fail.
+ *     </li>
+ *     <li>
+ *         Find an aggregate function in this package similar to the one you are working on and copy it to build
+ *         yours.
+ *         Your function might extend from the available abstract classes. Check the javadoc of each before using them:
+ *         <ul>
+ *             <li>
+ *                 {@link org.elasticsearch.xpack.esql.expression.function.aggregate.AggregateFunction}: The base class for aggregates
+ *             </li>
+ *             <li>
+ *                 {@link org.elasticsearch.xpack.esql.expression.function.aggregate.NumericAggregate}: Aggregation for numeric values
+ *             </li>
+ *             <li>
+ *                 {@link org.elasticsearch.xpack.esql.expression.function.aggregate.SpatialAggregateFunction}:
+ *                 Aggregation for spatial values
+ *             </li>
+ *         </ul>
+ *     </li>
+ *     <li>
+ *         Fill the required methods in your new function. Check their JavaDoc for more information.
+ *         Here are some of the important ones:
+ *         <ul>
+ *             <li>Constructor: Review the constructor annotations, and make sure to add the correct types and descriptions</li>
+ *             <li>
+ *                 {@code resolveType}: Check the metadata of your function parameters.
+ *                 This may include types, if they are foldable, or their possible values.
+ *             </li>
+ *             <li>
+ *                 {@code dataType}: This will return the datatype of your function.
+ *                 May be based on its current parameters.
+ *             </li>
+ *         </ul>
+ *
+ *         Finally, you may want to implement some interfaces.
+ *         Check their JavaDocs to see if they are suitable for your function:
+ *         <ul>
+ *             <li>
+ *                 {@link org.elasticsearch.xpack.esql.planner.ToAggregator}: (More information about aggregators below)
+ *             </li>
+ *             <li>
+ *                 {@link org.elasticsearch.xpack.esql.expression.SurrogateExpression}
+ *             </li>
+ *         </ul>
+ *     </li>
+ *     <li>
+ *         To introduce your aggregation to the engine:
+ *         <ul>
+ *             <li>
+ *                 Add it to {@code org.elasticsearch.xpack.esql.planner.AggregateMapper}.
+ *                 Check all usages of other aggregations there, and replicate the logic.
+ *             </li>
+ *             <li>
+ *                 Add it to {@link org.elasticsearch.xpack.esql.io.stream.PlanNamedTypes}.
+ *                 Consider adding the {@code writeTo} and {@code readFrom} methods inside your function,
```

Comment:
FWIW, it's more normal to make a deserializing constructor rather than `readFrom` method. You don't have to - and you shouldn't if it isn't possible because constructors have rules and sometimes those are annoying. But if you can, we tend to do that.